pkg/kubectl/cmd/util/factory.go

```diff
@@ -165,6 +166,24 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			list, err := client.ThirdPartyResources(api.NamespaceDefault).List(api.ListOptions{})
+			if err != nil {
+				fmt.Printf("Failed to populate third party resources: %v", err)
+				return nil, nil
+			}
+			gvs, _, err := thirdpartyresourcedata.ExtractGroupVersionKind(list)
+			if err != nil {
+				fmt.Printf("Failed to get third party group/versions")
+				return nil, nil
+			}
+			registered.SetThirdPartyAPIGroupVersions(gvs)
```

Comment:
The trouble is that it is a global constant set in an `init` method where we really can't connect to a client (we have none of the relevant configs when the `init` method is called.)

I put it here, because this is as close to a Builder.init() method as there is...  I'm open to other suggestions that don't involve refactoring all of the builder logic.